src/CaptureClient/ApiEventIdProvider.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureClient/ApiEventIdProvider.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <memory>
+
+namespace orbit_capture_client {
+
+[[nodiscard]] std::unique_ptr<NameEqualityApiEventIdProvider>
```

Comment:
Most probably, we will wish to keep the owner of the IdProvider agnostic of the type. So it will have to store unique_ptr.

I think, there will eventually be a factory method returning a unique_ptr constructing the IdProvider, wraps it into cacher, etc. Not sure how similar it will be to the current Create.